src/datastation/dv_dataset_get_attributes.py

```diff
@@ -38,13 +40,22 @@ def main():
         required=False,
     )
 
+    add_batch_proccessor_args(parser, report=False)
     add_dry_run_arg(parser)
     args = parser.parse_args()
 
     dataverse_client = DataverseClient(config["dataverse"])
 
+    if args.pid is None:
+        result = dataverse_client.search_api().search(dry_run=args.dry_run)
+        pids = [record['global_id'] for record in result]
```

Comment:
When I understand `yield` correctly: `result` is a stream and each set of items yielded by search is reduced to 25 `pids` before reading the next. That looks good enough for memory usage. Sadly `pids` is not a stream to integrate likewise with batch processing. `len(pids)` in `process_pids` also spoils the soup: the pids of all datasets (over 5K reads for archaeology) are loaded before the batch processing starts.